waku/waku_rln_relay/group_manager/on_chain/group_manager.nim

```diff
@@ -353,16 +352,17 @@ proc startOnchainSync(g: OnchainGroupManager): Future[void] {.async.} =
 
   let ethRpc = g.ethRpc.get()
 
-  # the block chunk size decays exponentially with the number of blocks
-  # the minimum chunk size is 100
-  var blockChunkSize = 1_000_000
+  # static block chunk size
+  let blockChunkSize = 2_000
 
-  var fromBlock = if g.latestProcessedBlock.isSome():
+  var fromBlock = if g.latestProcessedBlock.isSome() and 
+                  g.latestProcessedBlock.get() > g.rlnContractDeployedBlockNumber.get():
     info "resuming onchain sync from block", fromBlock = g.latestProcessedBlock.get()
     g.latestProcessedBlock.get() + 1
   else:
-    info "starting onchain sync from scratch"
-    BlockNumber(0)
+    let deployedBlockNumber = g.rlnContractDeployedBlockNumber.get()
```

Comment:
related to my other comment, of having `rlnContractDeployedBlockNumber` as an Option.

here we are not handling if `isSome` so technically it could raise an unhandled exception? Perhaps having the field as mandatory also simplifies this.

Reply:
see https://github.com/waku-org/nwaku/pull/1955#discussion_r1307034035